app/javascript/entrypoints/two_factor_authentication.ts

```diff
@@ -0,0 +1,199 @@
+import * as WebAuthnJSON from '@github/webauthn-json';
+import axios, { AxiosError } from 'axios';
+
+import ready from '../mastodon/ready';
+
+import 'regenerator-runtime/runtime';
+
+type PublicKeyCredentialCreationOptionsJSON = Parameters<
+  (typeof WebAuthnJSON)['create']
+>[0]['publicKey'];
```

Comment:
Is there a way that could be made more readable?

Reply:
Unfortunately this type is not exported by the package, so we need to get it through the `WebauthnJSON` function types :(

Reply:
Wouldn't `type PublicKeyCredentialCreationOptionsJSON = WebAuthnJSON.CredentialCreationOptionsJSON['publicKey'];` be more readable?

Or just import `CredentialCreationOptionsJSON` and use `CredentialCreationOptionsJSON['publicKey']` directly where it is used.